source/agora/test/Base.d

```diff
@@ -104,12 +104,9 @@ public class TestNetwork : Network
         If after 10 query attempts they haven't connected
         to each other, throw an exception.
 
-        Params:
-            count = Expected number of nodes
-
     ***************************************************************************/
 
-    public void waitUntilConnected (size_t count)
+    public void waitUntilConnected ()
```

Comment:
Well the point of doing this is that, if for some reason the `apis` array is empty, this just passes.

Reply:
Why should it pass?

Reply:
You'll enter the `foreach` for `attempts`. Then the `foreach` on `this.apis` will not be executed because `this.apis.length == 0`. Then L132 will pass (0 == 0) and it returns.

Reply:
I mean the `assert(network.getDiscoveredNodes().length == NodeCount);` will fail. And then you get the assert message where it's used.

Reply:
But then you rely on the caller checking that you actually connected to all the nodes.